### ISSUE TITLE
Some extensions b*tch about not being able to write into the extensions directory

### ISSUE THREAD

**Issue by yangm97** [nix-community/nix-vscode-extensions#46]
Noticed this at the console:

```
Error while activating TypeScript/JavaScript: {"errno":-2,"code":"ENOENT","syscall":"stat","path":"/nix/store/lc0l4y05jrfpqr5j5h70pjchpx801h6y-vscode-extension-visualstudioexptteam-vscodeintellicode-1.2.30/share/vscode/extensions/visualstudioexptteam.vscodeintellicode/cache"}.
```

I suppose this extension in particular isn't following the best practices and stuff, but I wonder what would be a fix that doesn't depend on upstream.

**Comment by MobSenpai** [comment]
Yeah. The same is happening with this extension: https://open-vsx.org/extension/golf1052/base16-generator, I have used it to auto select various themes using base16 colors, but It is throwing errors, only If I install it this way. Otherwise if I install from extensions tab inside vscode, it works.

**Comment by billy4479** [comment]
Same for the [catppuccin theme](https://marketplace.visualstudio.com/items?itemName=Catppuccin.catppuccin-vsc). Changing accent color causes `Error: EROFS: read-only file system, open '/home/<user>/.vscode-oss/extensions/catppuccin.catppuccin-vsc/themes/mocha.json'`.

**Comment by AmeerTaweel** [comment]
I recently came across this [Reddit post](https://www.reddit.com/r/NixOS/comments/1b21fbt/i_made_a_nix_function_to_easily_sandbox_apps/). The post and comments mention [Nix-Bwrapper](https://github.com/xNaxdy/nix-bwrapper) and [NixPak](https://github.com/nixpak/nixpak) for sandboxing packages. Maybe we can wrap extensions that want to write to the extensions directory, and bind another writable directory in their write location.

**Comment by deemp** [comment]
@AmeerTaweel, have you tried to implement your solution? If yes, what were the outcomes?

**Comment by AmeerTaweel** [comment]
Sorry for the late reply @deemp.

I tried using both [Nix-Bwrapper](https://github.com/xNaxdy/nix-bwrapper) and [NixPak](https://github.com/nixpak/nixpak) to sandbox VSCode extensions, but I was unsuccessful.

I could not reproduce @yangm97's issue with `visualstudioexptteam.vscodeintellicode`.

I was able to reproduce @mobsenpai's issue with `golf1052.base16-generator`. However, both [Nix-Bwrapper](https://github.com/xNaxdy/nix-bwrapper) and [NixPak](https://github.com/nixpak/nixpak) are designed to sandbox binaries. And the extension only contains JS files, so I was out of luck...

However, I still believe this approach might work with VSCode extensions that contain binaries.
